### DISCUSSION TITLE
New language but file extension is already used for another one

### DISCUSSION THREAD

**Discussion by lkalabis** [XAMPPRocky/tokei#1099] [Q&A] [answered]
Hi guys,

I tried to add a new language to the project named: Apex. Its the programming language from the Salesforce platform. Its very similar to Java so the adoptions are easy.
The only problem is the file extension: 'cls'. The same file extension is already used in the VB6 language. 
When I tried to compile my changes I got the following warning.
```
cargo build
   Compiling tokei v13.0.0-alpha.1 (/Users/.../tokei)
warning: unreachable pattern
    --> /Users/.../tokei/target/debug/build/tokei-f5c432fa95aa688c/out/language_type.rs:6064:35
     |
6064 |                 | "frm" | "bas" | "cls" => Some(VB6),
     |                                   ^^^^^
     |
     = note: `#[warn(unreachable_patterns)]` on by default

warning: `tokei` (lib) generated 1 warning
    Finished dev [unoptimized + debuginfo] target(s) in 7.95s
```

As far as I understand this, it would not count/use the cls logic anymore, once it would detect an VB6 case. Am I right?
What is the best way to handle such a case?

**Comment by XAMPPRocky** [comment, marked as answer]
Thank you for your question, right now this is a limitation of how the code is setup, so there's nothing can be done about conflicting files right now, eventually I will rework that logic however to allow for conflicting extensions.
